src/botPage/view/blockly/blocks/tools/webhook.js

```diff
@@ -0,0 +1,168 @@
+        }
+        // Disconnect any children that don't belong.
+        for (let i = 0; i < this.itemCount_; i++) {
+            const connection = this.getInput(`ADD${i}`).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1) {
+                connection.disconnect();
+            }
+        }
+        this.itemCount_ = connections.length;
+        this.updateShape_();
+        // Reconnect any child blocks.
+        for (let i = 0; i < this.itemCount_; i++) {
+            Blockly.Mutator.reconnect(connections[i], this, `ADD${i}`);
+        }
+    },
+    /**
+     * Store pointers to any connected child blocks.
+     * @param {!Blockly.Block} containerBlock Root block in mutator.
+     * @this Blockly.Block
+     */
+    saveConnections(containerBlock) {
+        let itemBlock = containerBlock.getInputTargetBlock('STACK');
+        let i = 0;
+        while (itemBlock) {
+            const input = this.getInput(`ADD${i}`);
+            itemBlock.valueConnection_ = input && input.connection.targetConnection;
+            i++;
+            itemBlock = itemBlock.nextConnection && itemBlock.nextConnection.targetBlock();
+        }
+    },
+    /**
+     * Modify this block to have the correct number of inputs.
+     * @private
+     * @this Blockly.Block
+     */
+    updateShape_() {
+        if (this.itemCount_ && this.getInput('EMPTY')) {
+            this.removeInput('EMPTY');
+        } else if (!this.itemCount_ && !this.getInput('EMPTY')) {
+            this.appendDummyInput('EMPTY').appendField(translate('Empty payload'));
+        }
+        let i;
+        for (i = 0; i < this.itemCount_; i++) {
+            if (!this.getInput(`ADD${i}`)) {
+                const input = this.appendValueInput(`ADD${i}`);
+                if (i === 0) {
+                    input.appendField(translate('Payload:'));
+                }
+            }
+        }
+        // Remove deleted inputs.
+        while (this.getInput(`ADD${i}`)) {
+            this.removeInput(`ADD${i}`);
+            i++;
+        }
+    },
+    onchange: function onchange(ev) {
+        if (!this.workspace || this.isInFlyout || this.workspace.isDragging()) {
+            return;
+        }
+
+        if (ev.type === Blockly.Events.MOVE) {
+            for (let i = 0; i < this.itemCount_; i++) {
+                const currentBlock = this.getInputTargetBlock(`ADD${i}`);
+                if (currentBlock && currentBlock.type !== 'webhook_payload') {
+                    currentBlock.unplug(true);
+                }
+            }
+        }
+    },
+};
+
+Blockly.JavaScript.webhook = block => {
+    const url = expectValue(block, 'WEBHOOK_URL');
+    const payloads = new Array(block.itemCount_);
+    for (let i = 0; i < block.itemCount_; i++) {
+        payloads[i] = Blockly.JavaScript.valueToCode(block, `ADD${i}`, Blockly.JavaScript.ORDER_ATOMIC) || null;
+    }
+
+    if (!url || !payloads) {
```

Comment:
- `payloads` could be renamed to `payload` since we're only sending a single payload.
- `payload` could be empty so it wouldn't meet the condition here and `''` would be returned incorrectly.

Reply:
payload won't be empty, array will still be created if `itemCount_` is 0. 

`
const payload = new Array(block.itemCount_);
`